server/src/controllers/beneficiary.controllers.ts

```diff
@@ -0,0 +1,45 @@
+import Beneficiary from "../models/BeneficiaryModel";
+import { Request, Response } from 'express';
+
+const editBeneficiary = async (req: Request, res: Response): Promise<Response> => {
+    try {
+        const beneficiaryID = req.params.id; // Use params instead of body for ID
+        if (beneficiaryID) {
+            const result = await Beneficiary.findByIdAndUpdate(
+                beneficiaryID,
+                req.body,
+                { new: true }
+            );
+            if (!result) {
+                return res.status(404).send({ message: "Beneficiary not found" });
+            }
+            return res.status(200).json(result);
+        } else {
+            return res.status(404).send({ message: "Missing Beneficiary ID" });
+        }
+    } catch (err: any) { // using `any` for simplicity; you can also create a custom error type
+        console.error(err.message);
```

Comment:
This is kinda nitpicky I think, but see example here (https://github.com/ChangePlusPlusVandy/vac-net/blob/a9c4e9bab9f8ff75490e1b4e561bcb6290a39fbc/server/src/controllers/session.controllors.ts#L19) for dealign with errors so TS won't be angry bc this also isn't typesafe,